### PR TITLE
Checks results coming from Wanda

### PR DESCRIPTION
**Pull request by dottorblaster**
As above, this PR introduces checks results coming from Wanda.

## How was this tested?
Storybook, unit tests

### CODE REVIEW

**Comment by dottorblaster** [comment]
`+555` ATM, sorry for this folks :smiling_face_with_tear:

**Review by arbulu89 — COMMENTED** [review]
Hey @dottorblaster 
Long review hehe

Besides that, I find the next things missing. They can be worked in other PRs, but we should track them:
- Update cluster details page checks result summary box
- Improve the `running` view
- Error handling

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          setLoading(false);
+          setExecutionData(fetchedExecutionData);
+          setCatalog(fetchedCatalogData.items);
+        }
+      )
+      .catch((error) => {
+        setLoading(false);
+        logError(error);
+      });
+  }, [onExecutionFetch, onCatalogFetch, setExecutionData, setCatalog]);
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:58]: As a side note, our linter is a little broken, but we have a rule that enforce that the dependency are satisfied inside a hook, for example you can't manipulate anything that it's not in the dependency array. I'm not commenting this line in particular, but the best pratice could be taken here https://www.npmjs.com/package/eslint-plugin-react-hooks

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  const [catalog, setCatalog] = useState(null);
+  const [selectedCheck, setSelectedCheck] = useState(null);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    Promise.all([onExecutionFetch(executionID), onCatalogFetch()])
+      .then(
+        ([{ data: fetchedExecutionData }, { data: fetchedCatalogData }]) => {
+          setLoading(false);
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:49]: Maybe even a custom hook works, if we don't want to wire up the saga entirely, I think it's a good use case

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      >
+        {hosts &&
+          hosts.map((agentID, idx) => (
+            <HostResultsWrapper
+              key={idx}
+              hostname={hostnames.find(({ id }) => agentID === id)?.hostname}
+              reachable
+              unreachableMessage=""
+            >
+              {checks.map((checkID) => {
```
Comment (CDimonaco) [inline assets/js/components/ExecutionResults/ExecutionResults.jsx:119]: Maybe we can extract that logic, inside the arrow function, inside another component? Seems a little bit convoluted to follow

File: assets/js/components/ExecutionResults/ExecutionResults.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:49]: Maybe in the future we can consider [useSWR](https://swr.vercel.app/)

**Review by nelsonkopliku — COMMENTED** [review]
Tricky one :smile: 

`WANDA_URL=http://localhost:4001 iex -S mix phx.server` helped making it run, plus adjusting wanda to make them click.

Then playing a bit with the API I got this result.

checks-execution-page-wanda

With the right set of data I could get a clue of what the output was. 

We'll need to iterate on, as usual.

File: assets/js/components/ChecksResults/checksUtils.js [hunk 4]
```diff
@@ -29,3 +29,43 @@ export const getHostname =
...
+  );
+};
+
+export const getChecks = (checkResults) => {
+  return checkResults.map(({ check_id }) => check_id);
+};
+
+export const getHealth = (checkResults, checkID, agentID) => {
+  const checkResult = checkResults.find(({ check_id }) => check_id === checkID);
+  if (checkResult) {
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply assets/js/components/ChecksResults/checksUtils.js:55]: + 1 for using a guard here if possible

File: assets/js/components/ChecksResults/checksUtils.js [hunk 5]
```diff
@@ -29,3 +29,43 @@ export const getHostname =
...
+    const agentCheckResult = checkResult.agents_check_results.find(
+      ({ agent_id }) => agent_id === agentID
+    );
+
+    const failedExpectationEvaluations =
+      agentCheckResult?.expectation_evaluations.filter(
+        (expectationEvaluation) => 'message' in expectationEvaluation
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply assets/js/components/ChecksResults/checksUtils.js:62]: Also, you need to filter out the `expect_same`expectations till we don't have the new UI or they will show as passing always, and I'd rather to remove them from the count for the first iteration.

File: assets/js/components/ExecutionResults/ExecutionResultsPage.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    state.hostsList.hosts.map(({ id, hostname }) => ({ id, hostname }))
+  );
+  const cluster = useSelector((state) =>
+    state.clustersList.clusters.find((cluster) => cluster.id === clusterID)
+  );
+
+  return (
+    <ExecutionResults
+      clusterID={clusterID}
+      executionID={executionID}
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:20]: I checked Wanda code and it looks like they're not ordered https://github.com/trento-project/wanda/blob/main/lib/wanda/executions.ex#L47

Opening a PR for that if it's ok.

[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:20]: There you go https://github.com/trento-project/wanda/pull/93

[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:20]: > we won't have the latest execution ids

Absolutely agreed @arbulu89, I'll track the leftover as a subtask in the context of the same user story. We miss the piece where the dashboard actually asks Wanda about the executions.

File: assets/js/trento.jsx [hunk 7]
```diff
@@ -60,6 +61,10 @@ const App = () => {
                 path="clusters/:clusterID/checks/results"
                 element={<ChecksResults />}
               />
+              <Route
+                path="clusters/:clusterID/execution/:executionID"
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/trento.jsx:65]: As above, we miss that part for now. Changing the route anyway :+1:

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        logError(error);
+      });
+  }, [onExecutionFetch, onCatalogFetch, setExecutionData, setCatalog]);
+
+  if (loading) {
+    return <LoadingBox text="Loading checks execution..." />;
+  }
+
+  if (executionData?.status === 'running') {
+    return <LoadingBox text="Check execution currently running..." />;
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:65]: Yeah absolutely

File: assets/js/components/ExecutionResults/ExecutionResults.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:58]: @arbulu89 yeah but:

1. we don't wanna retrigger the effect, we just want the data fetched once
2. usually it's better to have any variable that you read inside the deps array. In this case we are only using the `set` functions to change the state and trigger a rerender, and we aren't actually reading these variables, so it's unnecessary to have them in the dependency array.

Also @CDimonaco you're right, we have to review the whole `eslint` config. I created a ticket to do this.

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+  clusterID,
+  executionID,
+  clusterName,
+  cloudProvider,
+  hostnames = [],
+  onExecutionFetch = getExecutionResult,
+  onCatalogFetch = getCatalog,
+  onCatalogRefresh = () => {},
+}) => {
+  const [loading, setLoading] = useState(false);
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:38]: I always did it the other way around, it's the first time someone makes me notice this :D

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+          The following results are valid for on-premise bare metal platforms.
+          <br />
+          If you are running your HANA cluster on a different platform, please
+          use results with caution
+        </WarningBanner>
+      )}
+      <ResultsContainer
+        catalogError={false}
+        clusterID={clusterID}
+        hasAlreadyChecksResults
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:107]: This is a shorthand for `hasAlreadyChecksResults={true}` :)))

**Comment by dottorblaster** [comment]
@arbulu89 @fabriziosestito @CDimonaco @nelsonkopliku I think I addressed it all, could you see if I missed something?

File: assets/js/components/ExecutionResults/ExecutionResults.stories.jsx [hunk 11]
```diff
@@ -0,0 +1,154 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/ExecutionResults.stories.jsx:1]: I will! I promise :sweat_smile:

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  const checkResults = getCheckResults(executionData);
+  const hosts = getHosts(checkResults);
+  const checks = getChecks(checkResults);
+
+  return (
+    <div>
+      <Modal
+        title={catalog?.find(({ id }) => id === selectedCheck)?.description}
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:80]: We should use the new `getCheckDescription` here

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+        open={modalOpen}
+        onClose={() => setModalOpen(false)}
+      >
+        <ReactMarkdown className="markdown" remarkPlugins={[remarkGfm]}>
+          {catalog?.find(({ id }) => id === selectedCheck)?.description}
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/ExecutionResults.jsx:85]: The same, use `getCheckDescription` here

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                  health,
+                  expectations,
+                  failedExpectations
+                );
+
+                return (
+                  <CheckResult
+                    key={checkID}
+                    checkId={checkID}
+                    description={
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/ExecutionResults.jsx:140]: `getCheckDescription`

**Review by arbulu89 — APPROVED** [review]
@dottorblaster So far so good.
I think we can merge and iterate on the evolution!

File: assets/js/components/ChecksResults/checksUtils.js [hunk 15]
```diff
@@ -29,3 +29,47 @@ export const getHostname =
...
+    .filter(({ type }) => type !== 'expect');
+
+  return {
+    expectations: checkResult.expectation_results.length,
+    failedExpectations: failedExpectationEvaluations.length,
+    health: failedExpectationEvaluations.length > 0 ? 'critical' : 'passing',
+  };
+};
+
+export const getCheckDescription = (catalog, checkID) =>
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ChecksResults/checksUtils.js:74]: No, you're actually right

**Review by nelsonkopliku — APPROVED** [review]
Looks like a good starting point to iterate on, indeed.

File: assets/js/components/ChecksResults/checksUtils.js [hunk 16]
```diff
@@ -29,3 +29,47 @@ export const getHostname =
...
+  return checkResults.flatMap(({ agents_check_results }) =>
+    agents_check_results.map(({ agent_id }) => agent_id)
+  );
+};
+
+export const getChecks = (checkResults) => {
+  return checkResults.map(({ check_id }) => check_id);
+};
+
+export const getHealth = (checkResults, checkID, agentID) => {
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ChecksResults/checksUtils.js:53]: I hate that too :smile: looking forward to a new UX!

**Pull request merged**